ui/app/models/azure/config.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import Model, { attr } from '@ember-data/model';
+import fieldToAttrs, { expandAttributeMeta } from 'vault/utils/field-to-attrs';
+
+// Note: while the API docs indicate subscriptionId and tenantId are required, the UI does not enforce this because the user may pass these values in as environment variables.
+// https://developer.hashicorp.com/vault/api-docs/secret/azure#configure-access
+export default class AzureConfig extends Model {
+  @attr('string') backend; // dynamic path of secret -- set on response from value passed to queryRecord
+  @attr('string', {
+    label: 'Subscription ID',
+  })
+  subscriptionId;
+  @attr('string', {
+    label: 'Tenant ID',
+  })
+  tenantId;
+  @attr('string', {
+    label: 'Client ID',
+  })
+  clientId;
+  @attr('string', { sensitive: true }) clientSecret; // obfuscated, never returned by API
+  @attr('string', {
+    subText:
+      'The audience claim value for plugin identity tokens. Must match an allowed audience configured for the target IAM OIDC identity provider.',
+  })
+  identityTokenAudience;
+  @attr({
+    label: 'Identity token TTL',
+    helperTextDisabled:
+      'The TTL of generated tokens. Defaults to 1 hour, turn on the toggle to specify a different value.',
+    helperTextEnabled: 'The TTL of generated tokens.',
+    editType: 'ttl',
```

Comment:
ttl's can be so weird, and every API endpoint handles differently...I cant remember what the ttl component submits by default now, but does it mess with the backend default at all? For example, does it submit `0` or just nothing if it's not edited?

Reply:
nothing if it's not changed. Only if you toggle does it start to care. And in this case it sends 0s, which makes sense because you didn't fill out the field. 
<img width="540" alt="image" src="https://github.com/user-attachments/assets/0eb97b91-7263-4ee1-998b-5f0bb35c5b30">

I think there is some work we can do on the TTL to handle this better, maybe to show 0s instead of something empty, but this is the current behavior. Again, only if you toggle does it send something.

Reply:
Yep that sounds right. TTL inconsistencies are a whole other can of worms...just wanted to make sure we double checked the behavior here. Thanks!